src/shared/components/query/quickSearch/QuickSearch.tsx

```diff
@@ -165,6 +178,11 @@ export default class QuickSearch extends React.Component {
                         value: parseInt(response[5].headers["total-count"]) - patientOptions.length,
                         type: OptionType.PATIENT_COUNT
                     }
+                    const sampleOptions: any = response[6].body.map(this.sampleToOption);
```

Comment:
i can see this is an issue the predates this PR but i'm curious why this stuff isn't typed.    like, what is the actual return type of the client methods?

Reply:
There are two separate problems here: on the line you selected, the type of the response is not the reason `sampleOptions` is typed as any. Rather, the option type used by `studyToOption, geneToOption, patientToOption, sampleToOption` is super loosely typed. It's loosely typed because each of those use cases is unique. I could use union types to try and type it, but I'm not sure what the end result would be. If you want me to, I will. I'll probably try anyways because type systems are fun to fool around with.

The other problem is with the response type itself. You can type the response in the `then` as `[request.Response, request.Response, request.Response, request.Response, request.Response, request.Response, request.Response, request.Response, request.Response]` and that part compiles. Unfortunately, the type declaration seems a bit incorrect, because it thinks it should be `response[5].header` and not `response[5].headers`. Headers clearly works, so I'm blaming `node_modules/@types/superagent/index.d.ts#Response`. I can fix this by declaring an interface that extends `request.Response` that adds the `headers` field. However, `headers` would have to be of type `any` or have an index signature. I don't think there is a lot to gain here. Furthermore, upon looking at the actual `Object.keys(reponse[0])`, the interface I'd be extending is pretty far off, and it would be misleading to reference it. The console says that the Response is just `["body", "text", "headers", "statusCode", "status", "ok"]`, but the type declaration thinks it should have all of these fields:
```
        text: string;
        body: any;
        files: any;
        header: any;
        type: string;
        charset: string;
        status: number;
        statusType: number;
        info: boolean;
        ok: boolean;
        redirect: boolean;
        clientError: boolean;
        serverError: boolean;
        error: Error;
        accepted: boolean;
        noContent: boolean;
        badRequest: boolean;
        unauthorized: boolean;
        notAcceptable: boolean;
        notFound: boolean;
        forbidden: boolean;
        xhr: XMLHttpRequest;
        get(header: string): string;
        redirects: string[];
``` 
I think the existing `any`s are the best solution here.

Reply:
Update: I typed the Options so you can see what it looks like. The changes are in their own commit so you can read through them easily. I like static types a lot, so I am not an impartial observer.

If you like the changes I will squash them and keep them. If you don't like them, that's fine too- I can throw them out just as easily.

Reply:
I took a look at generated client and think i see what's happening.  All methods have accompanying WithHTTPInfo siblings.  These sibling method allow you to access the http headers, but for reason they do NOT type the response data (as the plain methods do).  You can see this looking at getAllGenesUsingGET and getAllGenesUsingGETWithHTTPInfo.    I think this is a crappy detail of the ts client generator.   Anyhow, lets not worry about any of this.  You've improved the situation with the Option type.  Thanks!